lib/generators/deimos/schema_class/templates/schema_record.rb.tt

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# autogenerated by Deimos, do not modify
+module Deimos
+  # :nodoc:
+  class <%= schema_classname(@current_schema) %> < SchemaRecord
+<%- fields.each do |field| -%>
+    # @return [<%= deimos_field_type(field) %>]
+    attr_accessor :<%= field.name %>
+<% end -%>
+<%- unless @current_schema.name =~ /\-key$/ -%>
+    # @return [Object] An optional payload key
+    attr_accessor :payload_key
+<% end -%>
+
+    # @override
+    def initialize(**kwargs)
+      super()
+      args = kwargs.with_indifferent_access
+<%- fields.each do |field| -%>
+      <%= "@#{field.name} = args[:#{field.name}]" %>
+<% end -%>
+      @payload_key = args[:payload_key]
+    end
+
+    # @override
+    def self.initialize_from_payload(payload)
+      return unless payload.present?
+
+      args = {}
+      payload.each do |key, value|
+<%- unless @special_field_initialization.present? -%>
+        args[key.to_sym] = value
+<%- else -%>
```

Comment:
Can you do `if/else` instead of `unless/else`?

Reply:
Yeap